libsolidity/parsing/Scanner.cpp

```diff
@@ -607,10 +642,19 @@ bool Scanner::scanEscape()
 	case 'v':
 		c = '\v';
 		break;
+	case 'u':
+	{
+		unsigned codepoint;
+		if (!scanUnicode(&codepoint))
+			return false;
+		addUnicodeChar(c);
```

Comment:
This is a bit confusing. Shouldn't it be `addUnicodeChar(codepoint);`? `c` just equals `'u'` at this point, doesn't it? Sorry if I simply understand the code wrong.